server/prometheus.go

```diff
@@ -92,33 +145,10 @@ func (pp *PrometheusProvider) execute(ctx *gin.Context) {
 	}
 	graph := row.getGraph(graphName)
 	if graph != nil {
-		fmt.Println(graph.QueryExpression)
-		tmpl, err := template.New("query").Parse(graph.QueryExpression)
-		if err != nil {
-			ctx.JSON(http.StatusBadRequest, "invalid query")
-			return
-		}
-		env1 := make(map[string]string)
-		for k, v := range env {
-			fmt.Println(k, v)
-			env1[k] = strings.Join(v, ",")
-		}
-		buf := new(bytes.Buffer)
 
-		err = tmpl.Execute(buf, env1)
-		if err != nil {
-			fmt.Println(err)
-			ctx.JSON(http.StatusBadRequest, err)
-			return
-		}
-		strQuery := buf.String()
-		r := v1.Range{
-			Start: time.Now().Add(-duration),
-			End:   time.Now(),
-			Step:  time.Minute,
-		}
-		fmt.Println(strQuery)
-		result, warnings, err := pp.provider.QueryRange(ctx, strQuery, r)
+		var data AggregatedResponse
+		result, warnings, err := ExecuteGraphQuery(ctx, graph.QueryExpression, env, duration, pp)
+
 		if err != nil {
 			fmt.Printf("Warnings: %v\n", warnings)
 			ctx.JSON(http.StatusBadRequest, err)
```

Comment:
All returned errors should have proper message added. See example in my comment above.

Reply:
Done the changes